sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/TestProxyPlaybackClient.java

```diff
@@ -0,0 +1,128 @@
+    private static final List<TestProxySanitizer> DEFAULT_SANITIZERS = loadSanitizers();
+    private final List<TestProxySanitizer> sanitizers = new ArrayList<>();
+
+    private static final List<TestProxyMatcher> DEFAULT_MATCHERS = loadMatchers();
+
+    private final List<TestProxyMatcher> matchers = new ArrayList<>();
+
+    /**
+     * Create an instance of {@link TestProxyPlaybackClient} with a list of custom sanitizers.
+     *
+     * @param customSanitizers the list of custom sanitizers to be added to {@link TestProxyPlaybackClient}
+     */
+    public TestProxyPlaybackClient(List<TestProxySanitizer> customSanitizers, List<TestProxyMatcher> customMatcher) {
+        this.sanitizers.addAll(DEFAULT_SANITIZERS);
+        this.sanitizers.addAll(customSanitizers == null ? Collections.emptyList() : customSanitizers);
+        this.matchers.addAll(DEFAULT_MATCHERS);
+        this.matchers.addAll(customMatcher == null ? Collections.emptyList() : customMatcher);
+    }
+
+    /**
+     * Starts playback of a test recording.
+     * @param recordFile The name of the file to read.
+     * @return A {@link Queue} representing the variables in the recording.
+     * @throws RuntimeException if an {@link IOException} is thrown.
+     */
+    public Queue<String> startPlayback(String recordFile) {
+        HttpRequest request = new HttpRequest(HttpMethod.POST, String.format("%s/playback/start", TestProxyUtils.getProxyUrl()))
+            .setBody(String.format("{\"x-recording-file\": \"%s\"}", recordFile));
+        try (HttpResponse response = client.sendSync(request, Context.NONE)) {
+            xRecordingId = response.getHeaderValue("x-recording-id");
+            addProxySanitization();
+            addMatcherRequests();
+            String body = response.getBodyAsString().block();
+            // The test proxy stores variables in a map with no guaranteed order.
+            // The Java implementation of recording did not use a map, but relied on the order
+            // of the variables as they were stored. Our scheme instead sets an increasing integer
+            // the key. See TestProxyRecordPolicy.serializeVariables.
+            // This deserializes the map returned from the test proxy and creates an ordered list
+            // based on the key.
+            return SERIALIZER.<Map<String, String>>deserialize(body, Map.class, SerializerEncoding.JSON).entrySet().stream().sorted(Comparator.comparingInt(e -> Integer.parseInt(e.getKey()))).map(Map.Entry::getValue).collect(Collectors.toCollection(LinkedList::new));
+
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Stops playback of a test recording.
+     */
+    public void stopPlayback() {
+        HttpRequest request = new HttpRequest(HttpMethod.POST, String.format("%s/playback/stop", TestProxyUtils.getProxyUrl()))
+            .setHeader("x-recording-id", xRecordingId);
+        client.sendSync(request, Context.NONE);
```

Comment:
That's not how the proxy works. The file isn't returned over the wire. Individual requests are answered. `playback/start` signals that we are going to start sending requests that should match a playback file. Then the test sends requests as it would in a live test and the proxy responds with what is in the file. `playback/stop` is so the proxy can free things for that test when it is done.